src/model/messages/mrtd/EMrtdDataSubmitMessage.ts

```diff
@@ -99,9 +99,12 @@ export class EMrtdDataSubmitMessage extends BaseMessage {
           parsed.fields.additionalPersonalData?.nameOfHolder ??
           `${parsedMrz.fields.lastName} ${parsedMrz.fields.firstName}`,
         dateOfBirth:
-          parsed.fields.additionalPersonalData?.fullDateOfBirth.toString().slice(2) ??
-          parsedMrz.fields.birthDate ??
-          undefined, // TODO: Check and specify date format
+          parsed.fields.additionalPersonalData?.fullDateOfBirth &&
+          parsed.fields.additionalPersonalData?.fullDateOfBirth !== 0
+            ? parsed.fields.additionalPersonalData.fullDateOfBirth.toString().slice(2)
+            : parsedMrz.fields.birthDate && parsedMrz.fields.birthDate !== '0'
+              ? parsedMrz.fields.birthDate
+              : undefined, // TODO: Check and specify date format
```

Comment:
Previously, an error occurred when the NFC received a value of `0`. The system processed this value and, upon performing the `slice`, treated it as valid but returned an empty string. This resulted in the date being interpreted as `null`, preventing the actual MRZ date from being assigned.
maybe something like:
```
const fullDateOfBirth = parsed.fields.additionalPersonalData?.fullDateOfBirth;
const birthDate = parsedMrz.fields.birthDate;
dateOfBirth: (fullDateOfBirth && fullDateOfBirth !== 0) 
  ? fullDateOfBirth.toString().slice(2) 
  : (birthDate && birthDate !== '0' ? birthDate : undefined);
```

Reply:
MRZ is being read from the eMRTD (equivalent to what you mention as 'NFC'). This value should not contain any error, since it is encoded in a file. Moreover, dates do have check digit, so I'm wondering in which cases you are receiving a 0. A log can be useful here.

I'm fine to make this code more error prone, but if there is an error while parsing the MRZ. it seems to me that the whole submission will be invalid. Do you have an example (a log or something) where this happened to you?

Reply:
You are correct that the default value of MRZ should always come with the proper values. I'm very sorry for the mistake here. The problem lies with the field `parsed.fields.additionalPersonalData?.fullDateOfBirth` of the eMRTD, as not all passports have the same fields in additionalPersonalData. This can cause errors. In my particular case, the `fullDateOfBirth` field of a Colombian passport was coming in as 0 (integer) and that's where the error occurred, preventing the date from being sent in the MRZ because the 0 with the string slice was sending an empty field.